icicle/include/icicle/program/symbol.h

```diff
@@ -0,0 +1,160 @@
+#pragma once
+
+#include <memory>
+#include <vector>
+
+namespace icicle {
+
+  /**
+   * @brief Enum to represent all the operation supported by the Symbol class.
+   *
+   */
+  enum OpCode {
```

Comment:
maybe once more namespace or rename? it's maybe a little too general and might conflict in the future